### DISCUSSION TITLE
Button Action to enter data in two different tables

### DISCUSSION THREAD

**Discussion by GuilhermeHirsch** [Budibase/budibase#2029] [Help] [answered]
I am trying, in the same action of a button, to insert data into two different tables (internal tables). 
A table created by me and the users (I created a column for this - "ID COBRANÇA"). 
My idea is to generate a temporary custom variable per user. 
However, nothing is saved in the users table and two rows are created in the table created by me.

![1](https://user-images.githubusercontent.com/33108596/126324046-211151c8-02fe-4739-9536-4c44d846ee74.jpg)

This column "ID COBRANÇA" is from the users table and was defined in action 3. It automatically appeared in action 2.

![2](https://user-images.githubusercontent.com/33108596/126324164-28a2fafb-1bc1-41da-8354-b40d115a5334.jpg)

**Comment by mjashanks** [comment, marked as answer]
Hey @GuilhermeHirsch .

You cannot insert rows into the user table in this way - we should not include this as an option. Users are managed centrally (outside of the app), so it doesn't really make sense.

Could you create another table, that links to users?

**Reply by GuilhermeHirsch** [reply]
I created another table related to the users table, but it had the same behavior: Duplicate row in one table, and in the other the data is not saved.

**Reply by GuilhermeHirsch** [reply]
And I noticed another strange behavior: Editing two columns manually at the action of a button. Once published, one of the columns disappears.

https://www.loom.com/share/084a459e5daa4435a53d7a8d6b249143

**Reply by GuilhermeHirsch** [reply]
The first situation I reported is in this video:

https://www.loom.com/share/fa373f98d56946f8b5a1be5d7c128db5

**Reply by mjashanks** [reply]
Thanks @GuilhermeHirsch, I'll look into these.

**Reply by mjashanks** [reply]
Your first issue is a bug. It seems like the 2nd `save row` is saving to the table defined in the form, rather than the table defined in the dropdown. I have logged this here: https://github.com/Budibase/budibase/issues/2039

You could create an automation that gets triggered on "Create Row" of your first table, and saves a record into your second table. We currently have a limitation in there, whereby we are unable to update Boolean or Dates in an automation action: https://github.com/Budibase/budibase/discussions/2028.

Your second issue is also a bug, but only a visual bug. You can see that if I click between other NON-`Save Row` actions, and back, then the correct values are displayed. Logged here, with video: https://github.com/Budibase/budibase/issues/2040

Thanks for reporting :)

**Reply by GuilhermeHirsch** [reply]
Wow! Excellent, @mjashanks! Thanks for the support. We move on 💪. Good work!
